sql/grant.go

```diff
@@ -40,12 +39,15 @@ func (p *planner) Grant(n *parser.Grant) (planNode, error) {
 		return nil, err
 	}
 
-	if !descriptor.HasPrivilege(p.user, parser.PrivilegeWrite) {
-		return nil, fmt.Errorf("user %s does not have %s privilege on %s %s",
-			p.user, parser.PrivilegeWrite, descriptor.TypeName(), descriptor.GetName())
+	if err := p.checkPrivilege(descriptor, privileges.WRITE); err != nil {
+		return nil, err
+	}
+
+	for _, grantee := range n.Grantees {
+		descriptor.GetPrivileges().Grant(grantee, n.Privileges)
```

Comment:
This is the auto-generated protobuf method, also why PrivilegeDescriptor is nullable.
There are other ways to do it, but all of them involve adding a bunch of extra functions to the interface and implementing them all on DatabaseDescriptor and TableDescriptor.